atmcfg/atmcfg_test.go

```diff
@@ -140,6 +171,20 @@ func TestAddUser(t *testing.T) {
 	}
 }
 
+func TestAddIndexConfig(t *testing.T) {
+	config := automationConfigWithIndexConfig()
+	u := indexConfig()
+	AddIndexConfig(config, u)
+	if len(config.IndexConfigs) != 2 {
+		t.Error("index not added\n")
+	}
+
+	AddIndexConfig(config, nil)
```

Comment:
in this scenario, the config adds nill as 3rd element of the array

Reply:
what I meant is that you use `&opsmngr.AutomationConfig{}` as input, as in with no initial values for `IndexConfigs` which may be nil, also you have a test dependency here, let's use
```golang
t.Run("name", func(t *testing.T){})
```
to encapsulate and name our scenarios